### DISCUSSION TITLE
AddDashboa services.AddElsaDashboardBackend

### DISCUSSION THREAD

**Discussion by DNDevCode** [elsa-workflows/elsa-core#717] [Q&A]
Hi, 

I have an exception when call services.AddElsaDashboardBackend() : 

System.InvalidOperationException : 'No IConfigureOptions<>, IPostConfigureOptions<>, or IValidateOptions<> implementations were found.'

Thank you

**Comment by sfmskywalker** [comment]
We'll need a little bit more detail than that. For example, what does your .csproj look like? Are you running one of the sample dashboard projects from the Elsa source code, or are you referencing Elsa packages? What .NET version are you using?

**Reply by DNDevCode** [reply]
Yes, 

- I checkout release/elsa-2.0
- Open projet Elsa.Samples.InvokeWorkflowFromController
- Add 2 packages nuget :  ElsaDashboard.Application and ElsaDashboard.Backend
- add service on startup :  services.AddElsaDashboardUI();  services.AddElsaDashboardBackend()
- Run 
System.InvalidOperationException at : services.AddElsaDashboardBackend() 
 
![image](https://user-images.githubusercontent.com/69428997/110217315-9640b180-7eb3-11eb-8c8d-ad0d6e4dddf8.png)

**Reply by sfmskywalker** [reply]
I think the issue is that you are adding package references instead of project references.

**Reply by DNDevCode** [reply]
I have removed all project references and i have the same issue : 

![image](https://user-images.githubusercontent.com/69428997/110220214-53d3a080-7ec4-11eb-9aa2-15585718ad3f.png)

Should i use project references ?

**Reply by DNDevCode** [reply]
And when i reference projects i got :

![image](https://user-images.githubusercontent.com/69428997/110220352-5da9d380-7ec5-11eb-8c6d-9311fff6ceb3.png)

Should i add some config options on appsetting ?

**Reply by sfmskywalker** [reply]
Can you submit a PR with your changes so I can take a look?

**Reply by DNDevCode** [reply]
https://github.com/elsa-workflows/elsa-core/pull/723

**Reply by DNDevCode** [reply]
Thank you for your time :)

**Reply by sfmskywalker** [reply]
Turns out I was wrong with the package references / project references.

Instead, you discovered a bug that causes this exception. When calling `AddElsaDashboardBackend` with no configurator, a call will be made to `services.ConfigureOptions<ElsaClientOptions>`, which fails because there are no implementations of the specified interfaces as per the exception message.

I pushed a fix that no longer calls `ConfigureOptions`. Instead relying on the hosting application to call `services.ConfigureOptions<T>` on whichever configurer type they want.

However, after you update your project with the fix, you'll encounter another issues, which is because the project `Elsa.Samples.HelloWorldHttp` does not have Blazor packages installed, but you are adding Elsa Dashboard services that require them. The Elsa Dashboard package is designed to be used for projects that have Blazor configured already.

**Reply by DNDevCode** [reply]
Thank you for the Fix. 

I have updated the project with the fix, and to try the fix i launch the project : ElsaDashboard.Samples.WebAssembly :

![image](https://user-images.githubusercontent.com/69428997/110326642-395f0b80-8019-11eb-87fb-6cb64a7159cf.png)

What is the default route to the Elsa designer ? still /elsa/home ? I received 404 error

**Reply by sfmskywalker** [reply]
Since the project represents a stand-along ASP.NET core Blazor app, the defaultroute is `/`.
You can check out `ElsaDashboard.Samples.Monolith` for an easy example. Alternatively, you can check out `ElsaDashboard.Samples.Server` for an example of a Dashboard that communicates with an external Elsa API server (which is represented by the `Elsa.Samples.Server.Host` project.

**Reply by DNDevCode** [reply]
OK, 
i want to encapsulate elsa in an api. i guess it's possible to have elsa in a dotnet core api with elsa designer (blazor) in the same appilciation?

I tried to launch the Elsa Dashboard.Examples.Monolith example project, the designer is well displayed, however when I try to create new workflow definition (Create button from designer) it throw an exception 

Microsoft.JSInterop.JSException: Failed to fetch dynamically imported module: https://localhost:3956/_content/Elsa.Designer.Bindings.Blazor/elsa-workflows.js
TypeError: Failed to fetch dynamically imported module: https://localhost:3956/_content/Elsa.Designer.Bindings.Blazor/elsa-workflows.js
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Elsa.Designer.Bindings.Blazor.WorkflowEditor.OnAfterRenderAsync(Boolean firstRender) in C:\Projects\Elsa2.0 Official\elsa-core\src\designer\bindings\blazor\Elsa.Designer.Bindings.Blazor\WorkflowEditor.razor:line 20
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)
Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost: Error: Unhandled exception in circuit 'VGckmbOrHVTOIrZFdZeouLvyx_C1SEyqtDyx2-JrK9U'.

Any idea ?

**Reply by DNDevCode** [reply]
this happens without any modification made on the monolith example project
